CommonKit/Source/UserIndicators/UserIndicatorPresentationContext.swift

```diff
@@ -0,0 +1,42 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import UIKit
+
+/// The presentation context is used by `UserIndicatorViewPresentable`s to display content
+/// on the screen and it serves two primary purposes:
+///
+/// - abstraction on top of UIKit (passing context instead of view controllers)
+/// - immutable context passed at init with variable presenting view controller
+///   (e.g. depending on collapsed / uncollapsed iPad presentation that changes
+///   at runtime)
+public protocol UserIndicatorPresentationContext {
+    var indicatorPresentingViewController: UIViewController? { get }
```

Comment:
PR looks good to me 👍

Feel free to ignore this, but maybe that could property could simply be named `presentingViewController` as the `indicator` prefix doesn't add much to me.

Reply:
I actually did have it as `presentingViewController` before, but had UIViewControllers conforming to this, which was clashing with the default `presentingViewController` by UIKit. At the moment there are no view controllers conforming to this but still wonder if its safer to disambiguate?

Reply:
Gotcha, that sounds safer to leave it as is then.